staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -368,29 +366,29 @@ func newTestableController(config TestableConfig) *configController {
 	return cfgCtlr
 }
 
-func (cfgCtlr *configController) Run(stopCh <-chan struct{}) error {
-	defer utilruntime.HandleCrash()
-
-	// Let the config worker stop when we are done
-	defer cfgCtlr.configQueue.ShutDown()
+func (cfgCtlr *configController) Start(ctx context.Context) error {
+	go func() {
+		<-ctx.Done()
+		// Let the config worker stop when we are done
+		klog.Info("Shutting down API Priority and Fairness config worker")
+		cfgCtlr.configQueue.ShutDown()
```

Comment:
if context is passed in pre-canceled or is canceled as Start is running, what happens?

Reply:
This is the controller that consumes the APF configuration API objects and digests them for use by the feature. Once the context is canceled, this controller will stop reacting to those API objects.

This `Start` method will always do some log statements.

The earlier the context is canceled, the fewer times this controller digests the config objects.

This controller is initialized with a short-term usable data structure before any config objects are digested, since the apiserver has to go into service before this controller can get and digest the config objects.

Reply:
Moved it below the "Running ..." logging.